web-common/src/features/dashboards/filters/dimension-filters/DimensionFilter.svelte

```diff
@@ -24,54 +40,237 @@
   export let timeControlsReady: boolean | undefined;
   export let smallChip = false;
   export let onRemove: () => void;
+  export let onApplyInList: (values: string[]) => void;
   export let onSelect: (value: string) => void;
+  export let onApplyContainsMode: (inputText: string) => void = () => {};
   export let onToggleFilterMode: () => void;
 
-  let open = openOnMount && !selectedValues.length;
-  let searchText = "";
-  let allValues: string[] = [];
+  let open = openOnMount && !selectedValues.length && !inputText;
+  $: sanitisedSearchText = inputText?.replace(/^%/, "").replace(/%$/, "");
+  let curMode = mode;
+  let curSearchText = "";
+  let curExcludeMode = excludeMode;
 
   $: ({ instanceId } = $runtime);
 
-  $: searchValues = useDimensionSearch(
+  $: resetFilterSettings(mode, sanitisedSearchText);
+
+  $: checkSearchText(curSearchText);
+
+  let searchedBulkValues: string[] =
```

Comment:
Nit: Can you please add some sensible line breaks from lines 60 to 114? Without any line breaks, it's pretty overwhelming.